cmd/scollector/collectors/nexpose.go

```diff
@@ -0,0 +1,405 @@
+		Add(&md, "nexpose.site.risk_score", site.RiskScore, opentsdb.TagSet{"site": site.Name}, metadata.Gauge, metadata.Score, descRiskScore)
+		tags := opentsdb.TagSet{"site": site.Name}
+
+		assets, err := c.getSiteAssets(site.ID)
+		if err != nil {
+			return nil, err
+		}
+		for _, asset := range *assets {
+			now := time.Now()
+			last_scan := now.Unix() - (asset.LastScanDate / 1000)
+			var assetTags opentsdb.TagSet
+			// Set the AssetName to the hostname if we have it, otherwise the IP.
+			if asset.AssetName != "" {
+				assetTags = opentsdb.TagSet{"name": asset.shortName()}.Merge(tags)
+			} else {
+				assetTags = opentsdb.TagSet{"name": asset.AssetIP}.Merge(tags)
+			}
+			Add(&md, "nexpose.asset.risk_score", asset.RiskScore, assetTags, metadata.Gauge, metadata.Score, descRiskScore)
+			Add(&md, "nexpose.asset.vuln_count", asset.VulnCount, assetTags, metadata.Gauge, metadata.Vulnerabilities, descVulnCount)
+			Add(&md, "nexpose.asset.exploit_count", asset.ExploitCount, assetTags, metadata.Gauge, metadata.Vulnerabilities, descExploitCount)
+			Add(&md, "nexpose.asset.malware_count", asset.MalwareCount, assetTags, metadata.Gauge, metadata.Vulnerabilities, descMalwareCount)
+			Add(&md, "nexpose.asset.last_scan", last_scan, assetTags, metadata.Gauge, metadata.Second, descAssetLastScan)
+
+			site.VulnCount += asset.VulnCount
+			site.ExploitCount += asset.ExploitCount
+			site.MalwareCount += asset.MalwareCount
+		}
+
+		Add(&md, "nexpose.site.vuln_count", site.VulnCount, tags, metadata.Gauge, metadata.Vulnerabilities, descVulnCount)
+		Add(&md, "nexpose.site.exploit_count", site.ExploitCount, tags, metadata.Gauge, metadata.Vulnerabilities, descExploitCount)
+		Add(&md, "nexpose.site.malware_count", site.MalwareCount, tags, metadata.Gauge, metadata.Vulnerabilities, descMalwareCount)
+	}
+
+	activeScans, err := c.scanActivity()
+	for _, scan := range *activeScans {
+		if scan.Status == "running" {
+			Add(&md, "nexpose.scan.running", 1, opentsdb.TagSet{"site": siteNames[scan.SiteID]}, metadata.Gauge, metadata.Bool, descScanRunning)
+		}
+	}
+
+	return md, nil
+}
+
+const (
+	descScanRunning   = "Nexpose scan running."
+	descRiskScore     = "Risk score for a given site/device."
+	descVulnCount     = "Number of known vulnerabilities."
+	descExploitCount  = "Number of vulnerabilities exploitable via Metasploit. Subset of vuln_count."
+	descMalwareCount  = "Number of vulnerabilities susceptible to malware attacks. Subset of vuln_count."
+	descAssetLastScan = "How many seconds ago this asset was last scanned."
+)
+
+func (c *nexposeConnection) jsonRequest(method string, location string, v *url.Values) (*[]byte, error) {
+	req, _ := http.NewRequest(method, "https://"+c.Host+location, strings.NewReader(v.Encode()))
+
+	// Some methods require the cookie, some require the header, so we set both.
+	req.AddCookie(c.Cookie)
+	req.Header.Add("nexposeCCSessionID", c.SessionID)
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
```

Comment:
I know their api is garbage, but setting Content-Type on a GET should definitely be an error.